RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -345,6 +356,78 @@ void MahiFit::updateCov(const SampleMatrix& samplecov) const {
   nnlsWork_.covDecomp.compute(invCovMat);
 }
 
+float MahiFit::ccTime(const float itQ) const {
+  float ccTime_ = 0.;
+
+  // those conditions are now on data time slices, can be done on the fitted pulse i.e. using nlsWork_.ampVec.coeff(itIndex);
+
+  const int soi = nnlsWork_.tsOffset;
+
+  // Selecting energetic hits - (Energy in TS[3] and TS[4]) > 20 GeV
+  const bool cond1 =
```

Comment:
If I read correctly, if any of the conditions are false, you would like to return zero.
Why not just do the following?
`if(nnlsWork_.amplitudes.coeffRef(soi) + nnlsWork_.amplitudes.coeffRef(soi + 1U) < thEnergeticPulses_) return 0.;`
and similarly for the other two conditions?
Then you can move the initialization of ccTime to below these 3 conditions.